pkg/server/server_test.go

```diff
@@ -236,6 +237,17 @@ func TestServerWithMySQLDatastoreAndExplicitCredentials(t *testing.T) {
 	test.RunAllTests(t, ds)
 }
 
+func TestSingleflightCheckResolverDefault(t *testing.T) {
```

Comment:
This will save our bacon because we can make sure now that the entrypoint for `Server#Check` starts at a SingleflightCheckResolver. This test gives us the fidelity we would have liked to have had prior when we were constructing a LocalChecker per request and therefore couldn't easily determine the precise resolver entrypoint the server was going to make when evaluating a Check.

Reply:
Love this test, simple yet effective. Though I think it should be recontextualized to be about the composition rather than the 

In practice, I see this as renaming to `TestCheckResolverComposition` and then checking the type of the top-level delegate and it's chain of sub-delegates for both a cache and uncached server configuration.

Reply:
I'll hold a vote on what the name should be. Until that is concluded, I don't think this PR is an appropriate forum to discuss renaming.